lib/project_types/extension/features/argo_config.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module Extension
+  module Features
+    class ArgoConfig
+      CONFIG_FILE_NAME = 'extension.config.yml'
+
+      class << self
+        def parse_yaml(context, permitted_keys = [])
+          file_name = File.join(context.root, CONFIG_FILE_NAME)
+
+          return {} unless File.size?(file_name)
+
+          require 'yaml' # takes 20ms, so deferred as late as possible.
+          begin
+            config = YAML.load_file(file_name)
+
+            # `YAML.load_file` returns nil if the file is not empty
+            # but does not contain any parsable yml data, e.g. only comments
+            # We consider this valid
+            return {} if config.nil?
+
+            unless config.is_a?(Hash)
+              raise ShopifyCli::Abort, ShopifyCli::Context.message('core.yaml.error.not_hash', CONFIG_FILE_NAME)
+            end
+
+            config.transform_keys!(&:to_sym)
+            assert_valid_config(config, permitted_keys) unless permitted_keys.empty?
+
+            config
+          rescue Psych::SyntaxError => e
+            raise(
+              ShopifyCli::Abort,
+              ShopifyCli::Context.message('core.yaml.error.invalid', CONFIG_FILE_NAME, e.message)
+            )
+          end
+        end
+
+        private
+
+        def assert_valid_config(config, permitted_keys)
+          unpermitted = config.keys.select do |k|
```

Comment:
tiny nitpick: I would've named the var `unpermitted_keys` for consistency

Reply:
changed!